includes/class-wc-order-item.php

```diff
@@ -284,6 +292,14 @@ public function get_formatted_meta_data( $hideprefix = '_', $include_all = false
 				continue;
 			}
 
+			if ( strpos( $this->invoke_template, 'email-order-items' ) !== false ) {
+				$order = $this->get_order();
+				// Skip backorder meta for a completed orders.
+				if ( $order && 'completed' === $order->get_status() && 'Backordered' === $meta->key ) {
```

Comment:
String `'Backordered'` is translatable, and is also filterable. A simple solution would look a little like this:

```php
				$backordered = apply_filters( 'woocommerce_backordered_item_meta_name', __( 'Backordered', 'woocommerce' ), $this );

				if ( $order && 'completed' === $order->get_status() && $backordered === $meta->key ) {
```

However it would still be suboptimal, because of the dynamic nature of filtered properties (potentially, different text could be used for different products).

---

Possibly a better alternative, which I think would be sufficiently backwards compatible, would be to start using a private piece of meta (`'_backordered'`) to track this information: this could then be used to conditionally inject information via `woocommerce_order_item_get_formatted_meta_data` or some other suitable hook.